tests/tests/custom_scan.rs

```diff
@@ -569,8 +571,24 @@ fn is_numeric_fast_field_capable(mut conn: PgConnection) {
     "#
     .execute(&mut conn);
 
-    let (b, count) = "select assert(count(*), 8), count(*) from (select id from test where message @@@ 'beer' order by severity) x limit 8;".fetch_one::<(bool, i64)>(&mut conn);
+    let sql = "select assert(count(*), 8), count(*) from (select id from test where message @@@ 'beer' order by severity) x limit 8;";
+    let (b, count) = sql.fetch_one::<(bool, i64)>(&mut conn);
     assert_eq!((b, count), (true, 8));
+
+    // This falls back to Normal because we fail to extract the relevant fast fields at execution
+    // time: see #TODO-ISSUE-NUMBER
+    let (plan,) =
+        format!("EXPLAIN (FORMAT JSON) {sql}").fetch_one::<(serde_json::Value,)>(&mut conn);
+    eprintln!(">>> {plan:#?}");
+    assert_eq!(
+        plan.pointer("/0/Plan/Plans/0/Plans/0/Plans/0")
+            .unwrap()
+            .as_object()
+            .unwrap()
+            .get("Exec Method")
+            .unwrap(),
+        &Value::String("NumericFastFieldExecState".to_owned())
+    );
```

Comment:
So, interestingly: this test was not actually using `NumericFastFieldExecState` before #2513, since we failed to detect anything except junk fields. So the test has been misnamed for a while.

But without the fallback to `Normal` (see my other comment) we currently fail to find any interesting fast fields in the target list at runtime.